### PR TITLE
start addressing sphinx warnings and rendering issues

### LINKED ISSUE #713: Create `numpydoc_xref_alias` for docstring aliases
The docstring can make use the aliases for more precise references

[PyMC Example doc of `tensor_like`](https://www.pymc.io/projects/docs/en/stable/api/distributions/generated/pymc.AsymmetricLaplace.html)
[PyMC conf.py for reference](https://github.com/pymc-devs/pymc/blob/main/docs/source/conf.py#L63-L80)

### LINKED ISSUE #727: Fix `ParetoNBDModel` Docstring Typos
While working on https://github.com/pymc-labs/pymc-marketing/issues/527, I noticed some docstring typos and syntax errors for `ParetoNBDModel`. The docstrings in `BetaGeoModel` follow an identical convention and are error-free, so this would be a very quick PR fix.

### PR DESCRIPTION
**Pull request by OriolAbril**
## Description

I went over documentation related issues to take care of while getting more familiar with the
library and tried to tackle some of them. I have fixed rendering issues and added numpydoc
xref links for types so far.

## Related Issue

- [x] Closes #713, also should close #727 (didn't check for typos in much depth though, but did fix
  all rendering issues already)
- [x] Related to #367

## Checklist

- [ ] Checked that [the pre-commit linting/style checks pass](https://docs.pymc.io/en/latest/contributing/python_style.html)
- [ ] Included tests that prove the fix is effective or that the new feature works
- [ ] Added necessary documentation (docstrings and/or example notebooks)
- [ ] If you are a pro: each commit corresponds to a [relevant logical change](https://wiki.openstack.org/wiki/GitCommitMessages#Structural_split_of_changes)

## Modules affected

- [x] CLV

## Type of change

- [ ] New feature / enhancement
- [ ] Bug fix
- [x] Documentation
- [x] Maintenance
- [ ] Other (please specify):

----
📚 Documentation preview 📚: https://pymc-marketing--750.org.readthedocs.build/en/750/

### CODE REVIEW

**Comment by wd60622** [comment]
Amazing!

File: pymc_marketing/clv/models/pareto_nbd.py [hunk 1]
```diff
@@ -1008,21 +1028,23 @@ def distribution_new_customer_recency_frequency(
...
             Number of observation periods for each customer. If not provided, T values from fit dataset will be used.
             Not needed if `data` parameter is provided with a `T` column.
-        random_seed : RandomState, optional
+        random_seed : ~numpy.random.RandomState, optional
             Random state to use for sampling.
 
         Returns
         -------
-        xr.Dataset
+        .Dataset
```
Comment (twiecki) [inline pymc_marketing/clv/models/pareto_nbd.py:1047]: xarray?

Comment (juanitorduz) [reply pymc_marketing/clv/models/pareto_nbd.py:1047]: +1

**Review by juanitorduz — APPROVED** [review]

**Comment by juanitorduz** [comment]
Thanks @OriolAbril ! Can we merge this one or would you like keep working on this branch ?

**Comment by OriolAbril** [comment]
Let's merge, it is better to do multiple PRs than to try fixing all rendering issues at once.

**Comment by juanitorduz** [comment]
Thanks @OriolAbril 🚀

**Pull request merged**